tests/bsim/bluetooth/mesh/src/test_provision.c

```diff
@@ -1354,6 +1424,147 @@ static void test_provisioner_pb_remote_client_ncrp_provision(void)
 	PASS();
 }
 
+/** @brief A device running a Remote Provisioning client and server that is used to reprovision
+ * another device and it self with the client.
+ */
+static void test_device_pb_remote_client_server_same_dev(void)
+{
+	NET_BUF_SIMPLE_DEFINE(dev_comp, BT_MESH_RX_SDU_MAX);
+	uint8_t status;
+	struct bt_mesh_cdb_node *node;
+	uint8_t page;
+	struct bt_mesh_key prev_dev_key;
+
+	k_sem_init(&prov_sem, 0, 1);
+	k_sem_init(&reprov_sem, 0, 1);
+
+	bt_mesh_device_setup(&prov, &rpr_cli_srv_comp);
+
+	ASSERT_OK(bt_mesh_cdb_create(test_net_key));
+	ASSERT_OK(bt_mesh_provision(test_net_key, 0, 0, 0, 0x0001, dev_key));
+
+	LOG_INF("Enabling PB-Remote server");
+	ASSERT_OK(bt_mesh_prov_enable(BT_MESH_PROV_REMOTE));
+
+	/* Provision a remote device with RPR Client and Server with local RPR Server. */
+	current_dev_addr = 0x0001;
+	struct bt_mesh_rpr_node srv = {
+		.addr = current_dev_addr,
+		.net_idx = 0,
+		.ttl = 3,
+	};
+
+	LOG_INF("Provisioner prov, waiting for prov ...\n");
+	ASSERT_OK(provision_remote(&srv, 1, &srv.addr));
+
+	ASSERT_OK(k_sem_take(&prov_sem, K_SECONDS(20)));
+
+	/* Check device key by adding bt_mesh_reprovision_remote appkey. */
+	ASSERT_OK(bt_mesh_cfg_cli_app_key_add(0, current_dev_addr, 0, 0, test_app_key, &status));
+	ASSERT_OK(status);
+
+	/* Swap callback to catch when device reprovisioned. */
+	prov.node_added = prov_node_added_rpr;
+
+	/* Reprovision a remote device with remote RPR Client and Server. */
+	current_dev_addr = 0x0002;
+	srv.addr = current_dev_addr;
+
+	/* Store initial Composition Data Page 0. */
+	ASSERT_OK(bt_mesh_cfg_cli_comp_data_get(0, current_dev_addr, 0, &page, &dev_comp));
+
+	node = bt_mesh_cdb_node_get(current_dev_addr);
+	ASSERT_TRUE(node);
+
+	LOG_INF("Refreshing 0x%04x device key ...\n", srv.addr);
+	reprovision_remote_devkey_client(&srv, node);
+
+	LOG_INF("Changing 0x%04x Composition Data ...\n", srv.addr);
+	reprovision_remote_comp_data_client(&srv, node, &dev_comp);
+
+	LOG_INF("Changing 0x%04x address ...\n", srv.addr);
+	reprovision_remote_address_client(&srv, node);
```

Comment:
Is it possible to put this into a for-loop with a test vector with addresses 0x0002 and 0x0001 to avoid code duplication?